lib/error_database.cpp

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2023 Pionix GmbH and Contributors to EVerest
+
+#include <utils/error_database.hpp>
+
+#include <utils/error.hpp>
+
+#include <everest/exceptions.hpp>
+#include <everest/logging.hpp>
+
+#include <fmt/format.h>
+
+namespace Everest {
+namespace error {
+
+ErrorHandle ErrorDatabase::add_error(std::shared_ptr<Error> error) {
+    BOOST_LOG_FUNCTION();
+
+    ErrorHandle handle(error->uuid);
+    if (this->has_error(handle)) {
+        EVTHROW(EverestErrorAlreadyExistsError(handle));
+    }
+    this->errors[handle] = error;
```

Comment:
Access to this map should be protected by a muted so we don't add/remove items whilst iterating over it